src/pytorch_lightning/plugins/precision/colossalai.py

```diff
@@ -0,0 +1,60 @@
+from typing import Any, Optional, Union
+
+from torch import Tensor
+from torch.optim import Optimizer
+
+import pytorch_lightning as pl
+from pytorch_lightning.plugins.precision.precision_plugin import PrecisionPlugin
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.warnings import WarningCache
+
+warning_cache = WarningCache()
+
+
+class ColossalAIPrecisionPlugin(PrecisionPlugin):
```

Comment:
it should accept precision as an argument which should be setup automatically inside accelerator_connector when users pass `Trainer(precision=16, strategy='collosalai')`.

Also does it support:
```
bf16
64-bit precision
```
?

if not, a MisconfigurationException should be raised for others.

Reply:
Done.

Reply:
Conceptual question here: It looks like we have overridden this precision plugin just so we can send out warnings and errors. A particular reason why it isn't handled within the strategy directly? Also, this leaves no room for customization, so I suggest to mark the precision plugin protected, signaling we don't intend a user to modify it. 
What do you think? cc @rohitgr7

Reply:
I think we should keep this because the optimizer handles the `backward` call and `clip_grad_by_norm`. Users should be able to provide a specific plugin for customization and still use this strategy, but it won't' be possible if we have these restrictions in the strategy itself.